lib/src/entrypoint.dart

```diff
@@ -62,21 +63,78 @@ class Entrypoint {
   // true.
   final String workingDir;
 
-  Package? _workspaceRoot;
+  static ({Package root, Package work}) _loadWorkspace(
+    String workingDir,
+    SystemCache cache,
+  ) {
+    if (!dirExists(workingDir)) {
+      fail('The directory `$workingDir` does not exist.');
+    }
+    // Keep track of all the pubspecs met when walking up the file system.
+    // The first of these is the workingPackage.
+    final pubspecsMet = <String, Pubspec>{};
+    for (final dir in parentDirs(workingDir)) {
+      final Pubspec pubspec;
+
+      try {
+        pubspec = Pubspec.load(
+          dir,
+          cache.sources,
+          containingDescription: RootDescription(dir),
+          allowOverridesFile: true,
+        );
+      } on FileException {
+        continue;
+      }
+      pubspecsMet[p.canonicalize(dir)] = pubspec;
+      final Package root;
+      if (pubspec.resolution == Resolution.none) {
+        root = Package.load(
+          null,
+          dir,
+          cache.sources,
+          alreadyLoadedPubspecs: pubspecsMet,
+          withPubspecOverrides: true,
+        );
+        for (final package in root.transitiveWorkspace) {
+          pubspecsMet.entries.first;
+          if (identical(pubspecsMet.entries.first.value, package.pubspec)) {
```

Comment:
I don't understand this?

Reply:
We want to find the `Package` object that has the first `Pubspec` that we loaded. Such that we can refer it.

Reply:
Are you relying on the iteration order for `.entries` -- that's seems a little surprising.

Maybe, make like a `firstPubspecMet` variable that is nullable?

Is this because the root loads the packages? Ah.